src/person/controllers/authenticate_controller.py

```diff
@@ -0,0 +1,118 @@
+import app.settings as api_settings
+from app.utils import failure_response
+from app.utils import success_response
+from django.contrib.auth import authenticate
+from django.contrib.auth import login
+from django.contrib.auth.models import User
+from django.utils import timezone
+from google.auth.transport import requests
+from google.oauth2 import id_token
+from person.models import Person
+from rest_framework import status
+from rest_framework.authtoken.models import Token
+
+
+class AuthenticateController:
+    def __init__(self, request, data, serializer):
+        self._request = request
+        self._data = data
+        self._serializer = serializer
+
+    def authenticate(self, username, password):
+        user = authenticate(username=username, password=password)
+        if user is not None:
+            login(self._request, user)
+            return user, status.HTTP_200_OK
+        else:
+            return None, status.HTTP_403_FORBIDDEN
```

Comment:
you should always be catching for errors first and then returning success at the end
code logic should follow something like
```
if user is None: do something
return success
```